exercises/practice/baffling-birthdays/.meta/example.ex

```diff
@@ -0,0 +1,55 @@
+defmodule BafflingBirthdays do
+  @moduledoc """
+  Estimate the probability of shared birthdays in a group of people.
+  """
+
+  @spec shared_birthday?(birthdates :: [Date.t()]) :: boolean()
+  def shared_birthday?(birthdates) do
+    result =
+      Enum.reduce_while(birthdates, MapSet.new(), fn date, birthdays ->
+        birthday = {date.month, date.day}
+
+        if MapSet.member?(birthdays, birthday) do
+          {:halt, :found_a_shared_birthday}
+        else
+          {:cont, MapSet.put(birthdays, birthday)}
+        end
+      end)
+
+    result == :found_a_shared_birthday
+  end
+
+  @spec random_birthdates(group_size :: integer()) :: [Date.t()]
+  def random_birthdates(group_size) do
+    for _ <- 1..group_size do
+      year = generate_non_leap_year_january_first(0, 3000)
```

Comment:
This is an interesting year range 😬 just so that we're clear: it's completely unnecessary to make the year random, right? To pass the tests, you might just as well choose the same non-leap year for all birthdates.

Reply:
Yeah, as far as the tests are concerned you could pick `year = 2025` and be done with it, that's true.
In terms of generating random `Date.t()`, this felt more natural to me. I guess it doesn't really matter?